MainRepository/Program.cs

```diff
@@ -1,7 +1,185 @@
+                  if (mGreenList.Count == 5) {
+                     mGameOver = true;
+                     mCount++;
+                     return;
+                  }
+                  if (!ValidWord ()) j--;
+                  else mList.Clear ();
+               }
+            }
+         }
+         mCount++;
+      }
+   }
+
+   /// <summary>Set the foreground color for the list</summary>
+   public char Color (char clr) {
+      if (mGreenList.Contains (clr)) Console.ForegroundColor = ConsoleColor.Green;
+      else if (mYellowList.Contains (clr))
+         Console.ForegroundColor = ConsoleColor.Yellow;
+      else Console.ForegroundColor = ConsoleColor.Gray;
+      return clr;
+   }
+
+   /// <summary>Displays the current state of the board</summary>
+   public void DisplayBoard (string word, bool isEntered = false) {
+      Console.CursorLeft = 0;
+      Console.CursorTop = 0;
+      Console.CursorVisible = false;
+      Console.WriteLine ();
+      Console.CursorLeft = (Console.WindowWidth / 2) - 5;
+      for (int i = 0; i < 6; i++) {
+         for (int j = 0; j < 5; j++) {
+            if (i < mRow || isEntered) {
+               if (word[j] == mWordle[i, j]) {
+                  Console.ForegroundColor = ConsoleColor.Green;
+                  if (!mGreenList.Contains (mWordle[i, j]))
+                     mGreenList.Add (mWordle[i, j]);
+               } else if (word.Contains (mWordle[i, j])) {
+                  Console.ForegroundColor = ConsoleColor.Yellow;
+                  if (!mYellowList.Contains (mWordle[i, j]))
+                     mYellowList.Add (mWordle[i, j]);
+               } else {
+                  Console.ForegroundColor = ConsoleColor.Gray;
+                  if (!mGrayList.Contains (mWordle[i, j]))
+                     mGrayList.Add (mWordle[i, j]);
+               }
+            }
+            Console.Write (mWordle[i, j] + "  ");
+            Console.ResetColor ();
+         }
+         Console.WriteLine ();
+         Console.WriteLine ();
+         Console.CursorLeft = (Console.WindowWidth / 2) - 5;
+      }
+      PrintAlphabet ();
+   }
+
+   /// <summary>Prints the alphabet</summary>
+   public void PrintAlphabet () {
+      Console.CursorLeft = (Console.WindowWidth / 2) - 13;
+      Console.WriteLine (new string ('_', 25));
+      Console.WriteLine ();
+      Console.CursorLeft = (Console.WindowWidth / 2) - 13;
+      for (int i = 0; i < 26; i++) {
+         char ch = (char)('A' + i);
+         Console.Write (Color (ch) + "   ");
+         if ((ch - 64) % 7 == 0) {
+            Console.SetCursorPosition ((Console.WindowWidth / 2) - 13, Console.CursorTop + 1);
+         }
+      }
+      Console.WriteLine ();
+      Console.CursorLeft = (Console.WindowWidth / 2) - 5;
+   }
+
+   /// <summary>Prints the final result</summary>
+   public void PrintResult (string word) {
+      Console.CursorLeft = (Console.WindowWidth / 2) - 13;
+      Console.WriteLine (new string ('_', 25));
+      if (mGameOver) {

```

Comment:
Both in if and else body similar code exists, refactor it.

Reply:
Both in if and else body similar code exists, refactor it. Refactor this as discussed.